[2022-02-09 17:54:22,731] {taskinstance.py:1032} INFO - Dependencies all met for <TaskInstance: spark-book-count.spark_job manual__2022-02-09T17:53:16.634887+00:00 [queued]>
[2022-02-09 17:54:22,751] {taskinstance.py:1032} INFO - Dependencies all met for <TaskInstance: spark-book-count.spark_job manual__2022-02-09T17:53:16.634887+00:00 [queued]>
[2022-02-09 17:54:22,751] {taskinstance.py:1238} INFO - 
--------------------------------------------------------------------------------
[2022-02-09 17:54:22,751] {taskinstance.py:1239} INFO - Starting attempt 2 of 2
[2022-02-09 17:54:22,751] {taskinstance.py:1240} INFO - 
--------------------------------------------------------------------------------
[2022-02-09 17:54:22,766] {taskinstance.py:1259} INFO - Executing <Task(SparkSubmitOperator): spark_job> on 2022-02-09 17:53:16.634887+00:00
[2022-02-09 17:54:22,771] {standard_task_runner.py:52} INFO - Started process 122 to run task
[2022-02-09 17:54:22,776] {standard_task_runner.py:76} INFO - Running: ['***', 'tasks', 'run', 'spark-book-count', 'spark_job', 'manual__2022-02-09T17:53:16.634887+00:00', '--job-id', '3', '--raw', '--subdir', 'DAGS_FOLDER/spark-dag.py', '--cfg-path', '/tmp/tmpqgy6rd4x', '--error-file', '/tmp/tmpp8r3o1cw']
[2022-02-09 17:54:22,776] {standard_task_runner.py:77} INFO - Job 3: Subtask spark_job
[2022-02-09 17:54:22,841] {logging_mixin.py:109} INFO - Running <TaskInstance: spark-book-count.spark_job manual__2022-02-09T17:53:16.634887+00:00 [running]> on host 6d5d9971f89c
[2022-02-09 17:54:22,943] {taskinstance.py:1426} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=***@***.com
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=spark-book-count
AIRFLOW_CTX_TASK_ID=spark_job
AIRFLOW_CTX_EXECUTION_DATE=2022-02-09T17:53:16.634887+00:00
AIRFLOW_CTX_DAG_RUN_ID=manual__2022-02-09T17:53:16.634887+00:00
[2022-02-09 17:54:22,963] {base.py:79} INFO - Using connection to: id: spark_default. Host:  spark://192.168.64.3:7077, Port: None, Schema: , Login: , Password: None, extra: {}
[2022-02-09 17:54:22,965] {spark_submit.py:360} INFO - Spark-Submit cmd: spark-submit --master  spark://192.168.64.3:7077 --conf spark.master=spark://spark:7077 --name Spark book count --verbose /usr/local/spark/app/spark-book-count.py /usr/local/spark/data/book.txt
[2022-02-09 17:54:24,816] {spark_submit.py:514} INFO - Using properties file: null
[2022-02-09 17:54:24,962] {spark_submit.py:514} INFO - WARNING: An illegal reflective access operation has occurred
[2022-02-09 17:54:24,962] {spark_submit.py:514} INFO - WARNING: Illegal reflective access by org.apache.spark.unsafe.Platform (file:/home/***/.local/lib/python3.7/site-packages/pyspark/jars/spark-unsafe_2.12-3.2.1.jar) to constructor java.nio.DirectByteBuffer(long,int)
[2022-02-09 17:54:24,962] {spark_submit.py:514} INFO - WARNING: Please consider reporting this to the maintainers of org.apache.spark.unsafe.Platform
[2022-02-09 17:54:24,962] {spark_submit.py:514} INFO - WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
[2022-02-09 17:54:24,963] {spark_submit.py:514} INFO - WARNING: All illegal access operations will be denied in a future release
[2022-02-09 17:54:25,046] {spark_submit.py:514} INFO - Parsed arguments:
[2022-02-09 17:54:25,046] {spark_submit.py:514} INFO - master                   spark://192.168.64.3:7077
[2022-02-09 17:54:25,046] {spark_submit.py:514} INFO - deployMode              null
[2022-02-09 17:54:25,047] {spark_submit.py:514} INFO - executorMemory          null
[2022-02-09 17:54:25,047] {spark_submit.py:514} INFO - executorCores           null
[2022-02-09 17:54:25,047] {spark_submit.py:514} INFO - totalExecutorCores      null
[2022-02-09 17:54:25,047] {spark_submit.py:514} INFO - propertiesFile          null
[2022-02-09 17:54:25,047] {spark_submit.py:514} INFO - driverMemory            null
[2022-02-09 17:54:25,047] {spark_submit.py:514} INFO - driverCores             null
[2022-02-09 17:54:25,047] {spark_submit.py:514} INFO - driverExtraClassPath    null
[2022-02-09 17:54:25,047] {spark_submit.py:514} INFO - driverExtraLibraryPath  null
[2022-02-09 17:54:25,047] {spark_submit.py:514} INFO - driverExtraJavaOptions  null
[2022-02-09 17:54:25,048] {spark_submit.py:514} INFO - supervise               false
[2022-02-09 17:54:25,048] {spark_submit.py:514} INFO - queue                   null
[2022-02-09 17:54:25,048] {spark_submit.py:514} INFO - numExecutors            null
[2022-02-09 17:54:25,048] {spark_submit.py:514} INFO - files                   null
[2022-02-09 17:54:25,048] {spark_submit.py:514} INFO - pyFiles                 null
[2022-02-09 17:54:25,048] {spark_submit.py:514} INFO - archives                null
[2022-02-09 17:54:25,048] {spark_submit.py:514} INFO - mainClass               null
[2022-02-09 17:54:25,048] {spark_submit.py:514} INFO - primaryResource         file:/usr/local/spark/app/spark-book-count.py
[2022-02-09 17:54:25,048] {spark_submit.py:514} INFO - name                    Spark book count
[2022-02-09 17:54:25,048] {spark_submit.py:514} INFO - childArgs               [/usr/local/spark/data/book.txt]
[2022-02-09 17:54:25,048] {spark_submit.py:514} INFO - jars                    null
[2022-02-09 17:54:25,049] {spark_submit.py:514} INFO - packages                null
[2022-02-09 17:54:25,049] {spark_submit.py:514} INFO - packagesExclusions      null
[2022-02-09 17:54:25,049] {spark_submit.py:514} INFO - repositories            null
[2022-02-09 17:54:25,049] {spark_submit.py:514} INFO - verbose                 true
[2022-02-09 17:54:25,049] {spark_submit.py:514} INFO - 
[2022-02-09 17:54:25,049] {spark_submit.py:514} INFO - Spark properties used, including those specified through
[2022-02-09 17:54:25,049] {spark_submit.py:514} INFO - --conf and those from the properties file null:
[2022-02-09 17:54:25,049] {spark_submit.py:514} INFO - (spark.master,spark://spark:7077)
[2022-02-09 17:54:25,049] {spark_submit.py:514} INFO - 
[2022-02-09 17:54:25,050] {spark_submit.py:514} INFO - 
[2022-02-09 17:54:25,124] {spark_submit.py:514} INFO - Exception in thread "main" org.apache.spark.SparkException: Master must either be yarn or start with spark, mesos, k8s, or local
[2022-02-09 17:54:25,124] {spark_submit.py:514} INFO - at org.apache.spark.deploy.SparkSubmit.error(SparkSubmit.scala:972)
[2022-02-09 17:54:25,125] {spark_submit.py:514} INFO - at org.apache.spark.deploy.SparkSubmit.prepareSubmitEnvironment(SparkSubmit.scala:238)
[2022-02-09 17:54:25,125] {spark_submit.py:514} INFO - at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:898)
[2022-02-09 17:54:25,125] {spark_submit.py:514} INFO - at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180)
[2022-02-09 17:54:25,125] {spark_submit.py:514} INFO - at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203)
[2022-02-09 17:54:25,125] {spark_submit.py:514} INFO - at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90)
[2022-02-09 17:54:25,125] {spark_submit.py:514} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1043)
[2022-02-09 17:54:25,125] {spark_submit.py:514} INFO - at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1052)
[2022-02-09 17:54:25,125] {spark_submit.py:514} INFO - at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
[2022-02-09 17:54:25,192] {taskinstance.py:1700} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1329, in _run_raw_task
    self._execute_task_with_callbacks(context)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1455, in _execute_task_with_callbacks
    result = self._execute_task(context, self.task)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1511, in _execute_task
    result = execute_callable(context=context)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 182, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 446, in submit
    f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {returncode}."
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master  spark://192.168.64.3:7077 --conf spark.master=spark://spark:7077 --name Spark book count --verbose /usr/local/spark/app/spark-book-count.py /usr/local/spark/data/book.txt. Error code is: 1.
[2022-02-09 17:54:25,212] {taskinstance.py:1277} INFO - Marking task as FAILED. dag_id=spark-book-count, task_id=spark_job, execution_date=20220209T175316, start_date=20220209T175422, end_date=20220209T175425
[2022-02-09 17:54:25,248] {standard_task_runner.py:92} ERROR - Failed to execute job 3 for task spark_job
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/task/task_runner/standard_task_runner.py", line 85, in _start_by_fork
    args.func(args, dag=self.dag)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/cli/cli_parser.py", line 48, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/cli.py", line 92, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/cli/commands/task_command.py", line 298, in task_run
    _run_task_by_selected_method(args, dag, ti)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/cli/commands/task_command.py", line 107, in _run_task_by_selected_method
    _run_raw_task(args, ti)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/cli/commands/task_command.py", line 184, in _run_raw_task
    error_file=args.error_file,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/session.py", line 70, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1329, in _run_raw_task
    self._execute_task_with_callbacks(context)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1455, in _execute_task_with_callbacks
    result = self._execute_task(context, self.task)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1511, in _execute_task
    result = execute_callable(context=context)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 182, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 446, in submit
    f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {returncode}."
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master  spark://192.168.64.3:7077 --conf spark.master=spark://spark:7077 --name Spark book count --verbose /usr/local/spark/app/spark-book-count.py /usr/local/spark/data/book.txt. Error code is: 1.
[2022-02-09 17:54:25,282] {local_task_job.py:154} INFO - Task exited with return code 1
[2022-02-09 17:54:25,344] {local_task_job.py:264} INFO - 0 downstream tasks scheduled from follow-on schedule check
